docker/generic/Dockerfile

```diff
@@ -0,0 +1,52 @@
+FROM alpine:3.7
+MAINTAINER Emmanuel Frecon <efrecon@gmail.com>
+
+LABEL Description="Eclipse Mosquitto MQTT Broker"
+
+ARG VERSION=1.4.15
+
+RUN set -ex \
+    # Add build dependencies, remove after build
+    && apk --no-cache add --virtual .build-deps \
+        build-base \
+        libressl-dev \
+        c-ares-dev \
+        util-linux-dev \
+        libwebsockets-dev \
+    # Add fetch dependencies, remove after build
+    && apk --no-cache add --virtual .fetch-deps \
+        git \
+        curl \
+    # Add run dependencies, keep after build
+    && apk --no-cache add --virtual .run-deps \
+        libressl \
+        c-ares \
+        util-linux \
+        libwebsockets \
+    # Checkout and compile, versions are branched with a "v" in front of the
+    # version number, but tagged at the Docker registry without the "v".
+    && git clone --depth 1 -b v${VERSION} https://github.com/eclipse/mosquitto.git /tmp/mosquitto \
+    && cd /tmp/mosquitto \
+    && curl -qsL https://git.alpinelinux.org/cgit/aports/plain/main/mosquitto/libressl.patch -o libressl.patch \
+    && git apply libressl.patch \
+    && make \
+        WITH_MEMORY_TRACKING=no \
```

Comment:
why are you turning this off?

Reply:
This is taken from what the Alpine people were doing when compiling mosquitto.

Reply:
Perhaps alpine is making poor choices?  This is somethign that used to be off in some distributions because uclibc didn't support it.  I believe musl does?  You seem to be simply cargo culting settings with no real basis for them though.

Reply:
You *do* understand that this is what the *official* (at least tagged as so, seen from most Docker users) image for [mosquitto](https://hub.docker.com/_/eclipse-mosquitto/) was compiled? This state of affairs was the reason why I did not want to depart from the current state. As I pointed out in my comment further down, this would mean that the project makes a decision about what the Docker image would contain, which is fine and which we can do here. Let me try this and the other switches that you pointed out and return to you.